File: Data/Invoice.cs
namespace SJC_Portal.Data
{
    public class Invoice
    {
        public int? InvoiceNumber { get; set; }
        public string? InvoiceDate { get; set; }
        public IEnumerable<BuildingDetails>? Buildings { get; set; }
        public BuildingDetails? ToBeInvoiced { get; set; }
        public double? InvoiceCharge { get; set; }
        public double? CraneCharge { get; set; }

    }
}

File: wwwroot/js/ProjectClass/Invoice.js
import { BuildingDetails } from "./BuildingDetails.js";
export class Invoice {
    InvoiceNumber;
    InvoiceDate;
    Buildings;
    ToBeInvoiced;
    InvoiceCharge;
    CraneCharge;

    constructor() {
        this.InvoiceNumber = "";
        this.InvoiceDate = "";
        this.Buildings = [new BuildingDetails()];
        this.Buildings.pop();
        this.ToBeInvoiced = new BuildingDetails();
        this.InvoiceCharge = 0;
        this.CraneCharge = 0;
    }
}